test/babylonFixtures.js

```diff
@@ -0,0 +1,55 @@
+import forEach from 'lodash.foreach';
+import get from 'lodash.get';
+import { multiple as getFixtures } from 'babel-helper-fixtures';
+import { parseAndGetProgram } from './utils';
+import xfail from './xfail.json';
+
+const fixtures = getFixtures(__dirname + '/babylon/test/fixtures');
+
+function runTest(task) {
+    const code = task.actual.code;
+    const throws = task.options.throws;
+    try {
+        parseAndGetProgram(task.actual.code);
+    } catch (err) {
+        if (throws) {
+            return;
+        }
+        throw err;
+    }
+
+    if (throws) {
+        throw new Error(`Expected error message: ${throws}`);
+    }
+}
+
+describe('babylon fixture tests', function() {
+    forEach(fixtures, function(suites, name) {
+        describe(`:: ${name}`, function() {
+            forEach(suites, function(testSuite) {
+                describe(`:: ${testSuite.title}`, function() {
+                    forEach(testSuite.tests, function(task) {
+                        const shouldXFail = get(xfail, [name, testSuite.title, task.title]);
+                        if (shouldXFail === 'skip') {
+                            // this test actually hangs the test suite!
+                            it.skip(`:: ${task.title}`, function() {});
+                            return;
+                        }
+                        it(`:: ${task.title}`, function() {
+                            if (shouldXFail) {
+                                try {
+                                    return runTest(task);
+                                } catch (e) {
+                                    this.skip();
+                                }
+                                throw Error('This test should fail!');
```

Comment:
If someone fixes a test case we will know about it, and we can remove it from the xfail.json